tests/pytorch_tests/model_tests/feature_models/const_quantization_test.py

```diff
@@ -0,0 +1,88 @@
+# limitations under the License.
+# ==============================================================================
+import torch
+import torch.nn as nn
+import numpy as np
+import model_compression_toolkit as mct
+from model_compression_toolkit.core.pytorch.utils import to_torch_tensor, torch_tensor_to_numpy, set_model
+from tests.pytorch_tests.model_tests.base_pytorch_feature_test import BasePytorchFeatureNetworkTest
+from tests.common_tests.helpers.tensors_compare import cosine_similarity
+from model_compression_toolkit.target_platform_capabilities.constants import IMX500_TP_MODEL
+from model_compression_toolkit.constants import PYTORCH
+from mct_quantizers import PytorchQuantizationWrapper
+
+tp = mct.target_platform
+
+
+class ConstQuantizationNet(nn.Module):
+    def __init__(self, layer, const):
+        super().__init__()
+        self.layer = layer
+        self.const = to_torch_tensor(const) if isinstance(const, np.ndarray) else const
+
+    def forward(self, x):
+        return self.layer(x, self.const)
+
+
+class ConstQuantizationReverseOrderNet(nn.Module):
+    def __init__(self, layer, const):
+        super().__init__()
+        self.layer = layer
+        self.const = to_torch_tensor(const) if isinstance(const, np.ndarray) else const
+
+    def forward(self, x):
+        return self.layer(self.const, x)
+
+
+class ConstQuantizationTest(BasePytorchFeatureNetworkTest):
+
+    def __init__(self, unit_test, func, const, input_reverse_order=False):
+        super().__init__(unit_test=unit_test, input_shape=(16, 32, 32))
+        self.func = func
+        self.const = const
+        self.input_reverse_order = input_reverse_order
+
+    def generate_inputs(self):
+        return [np.random.random(in_shape)+1 for in_shape in self.get_input_shapes()]
+
+    def get_tpc(self):
+        return mct.get_target_platform_capabilities(PYTORCH, IMX500_TP_MODEL, "v2")
+
+    # def get_quantization_config(self):
+    #     return mct.core.QuantizationConfig(mct.core.QuantizationErrorMethod.NOCLIPPING,
+    #                                        mct.core.QuantizationErrorMethod.NOCLIPPING,
+    #                                        False, False, True)
+
+    def create_networks(self):
+        if self.input_reverse_order:
+            return ConstQuantizationReverseOrderNet(self.func, self.const)
+        else:
+            return ConstQuantizationNet(self.func, self.const)
+
+    def compare(self, quantized_model, float_model, input_x=None, quantization_info=None):
+        in_torch_tensor = to_torch_tensor(input_x[0])
+        set_model(float_model)
+        y = float_model(in_torch_tensor)
+        y_hat = quantized_model(in_torch_tensor)
+        self.unit_test.assertTrue(y.shape == y_hat.shape, msg=f'out shape is not as expected!')
+        cs = cosine_similarity(torch_tensor_to_numpy(y), torch_tensor_to_numpy(y_hat))
+        self.unit_test.assertTrue(np.isclose(cs, 1, atol=1e-1), msg=f'fail cosine similarity check: {cs}')
```

Comment:
Same (CS). here it's even worse, if the tolerance needs to be 1e-1, are you sure that there isn't anything wrong with the results?
(by the way, I'm not sure how this test even works, given that you didn't add const quantization in the pytorch TPCs)